modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/elasticsearch/query/translator/factory/dsl/ExpressionQueryBuilder.java

```diff
@@ -0,0 +1,74 @@
+package com.enonic.xp.repo.impl.elasticsearch.query.translator.factory.dsl;
+
+import java.time.ZoneOffset;
+
+import org.elasticsearch.common.Strings;
+
+import com.enonic.xp.data.PropertySet;
+import com.enonic.xp.data.ValueTypes;
+import com.enonic.xp.repo.impl.elasticsearch.query.translator.resolver.SearchQueryFieldNameResolver;
+import com.enonic.xp.repo.impl.index.IndexValueType;
+
+abstract class ExpressionQueryBuilder
+    extends DslQueryBuilder
+{
+    private static final SearchQueryFieldNameResolver FIELD_NAME_RESOLVER = new SearchQueryFieldNameResolver();
+
+    protected final String type;
+
+    protected final String field;
+
+    ExpressionQueryBuilder( final PropertySet expression )
+    {
+        super( expression );
+
+        this.field = getString( "field" );
+        this.type = getString( "type" );
+    }
+
+    protected Object parseValue( final Object value )
+    {
+        if ( type == null || value == null )
+        {
+            return value;
+        }
+
+        switch ( type )
+        {
+            case "time":
+                return ValueTypes.LOCAL_TIME.convert( value );
+            case "date":
+                return ValueTypes.LOCAL_DATE.convert( value );
+            case "dateTime":
+                return ValueTypes.DATE_TIME.convert( value );
+            case "localDateTime":
+                return ValueTypes.LOCAL_DATE_TIME.convert( value ).toInstant( ZoneOffset.UTC );
```

Comment:
why we convert to UTC only localDateTime, and not localDate ? They should be handled equally.